src/ThreeDSAuth.res

```diff
@@ -0,0 +1,94 @@
+open Utils
+let getKeyValue = (json, str) => {
+  json
+  ->Js.Dict.get(str)
+  ->Belt.Option.getWithDefault(Js.Dict.empty()->Js.Json.object_)
+  ->Js.Json.decodeString
+  ->Belt.Option.getWithDefault("")
+}
+
+@react.component
+let make = () => {
+  let (openModal, setOpenModal) = React.useState(_ => false)
+  let (loader, setloader) = React.useState(_ => true)
+
+  let logger = Recoil.useRecoilValueFromAtom(RecoilAtoms.loggerAtom)
+
+  React.useEffect0(() => {
+    handlePostMessage([("iframeMountedCallback", true->Js.Json.boolean)])
+    let handle = (ev: Window.event) => {
+      let json = ev.data->Js.Json.parseExn
+      let dict = json->Utils.getDictFromJson
+      if dict->Js.Dict.get("fullScreenIframeMounted")->Belt.Option.isSome {
+        let metadata = dict->getJsonObjectFromDict("metadata")
+        let metaDataDict =
+          metadata->Js.Json.decodeObject->Belt.Option.getWithDefault(Js.Dict.empty())
+        let paymentIntentId = metaDataDict->getString("paymentIntentId", "")
+        let headersDict =
+          metaDataDict
+          ->getJsonObjectFromDict("headers")
+          ->Js.Json.decodeObject
+          ->Belt.Option.getWithDefault(Js.Dict.empty())
+        let threeDsAuthoriseUrl =
+          metaDataDict
+          ->getJsonObjectFromDict("threeDSData")
+          ->Js.Json.decodeObject
+          ->Belt.Option.getWithDefault(Js.Dict.empty())
+          ->getString("three_ds_authorize_url", "")
+        let headers =
+          headersDict
+          ->Js.Dict.entries
+          ->Js.Array2.map(entries => {
+            let (x, val) = entries
+            (x, val->Js.Json.decodeString->Belt.Option.getWithDefault(""))
+          })
+
+        let threeDsMethodComp = metaDataDict->getString("3dsMethodComp", "U")
```

Comment:
Can we please maintain parity in the naming convention of threeDS wherever it is being used?